lib/AST/NameLookup.cpp

```diff
@@ -2812,28 +2872,53 @@ createOpaqueParameterGenericParams(
     if (!typeRepr)
       continue;
 
-    auto opaqueTypeReprs = typeRepr->collectOpaqueReturnTypeReprs();
-    for (auto opaqueRepr : opaqueTypeReprs) {
+    // Plain protocols should imply 'some' with experimetal feature
+    CollectedOpaqueReprs typeReprs;
+    if (ctx.LangOpts.hasFeature(Feature::ImplicitSome)) {
+        typeReprs = collectTypeReprs(typeRepr);
+    } else {  typeReprs = collectOpaqueReturnTypeReprs(typeRepr); }
+
+      for (auto repr : typeReprs) {
+   
+        if (isa<IdentTypeRepr>(repr)){
+            if(!isProtocol(evaluator,repr, ctx, dc))
```

Comment:
Should we check for `isProtocol` inside of `collectOpaqueReturnTypeReprs` and only collect the `IdentTypeRepr` if it represents a protocol?

Reply:
updated!